bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitTaggedItem.java

```diff
@@ -214,12 +213,15 @@ public double getConfigurationAsDouble(String key, double defaultValue) {
     /**
      * parse and apply item configuration.
      */
+    @SuppressWarnings("null")
```

Comment:
Making the configuration final doesn't solve this warning?

Reply:
yes, still get this warning 
`[WARNING] /Users/efreiter/projects/knx/oh_addon/openhab-addons/bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitTaggedItem.java:[217,7459] Potential null pointer access: this expression has a '@Nullable' type`

the issue is the configuration. 
```
    private final @Nullable Map<String, Object> configuration;
...
  if (configuration != null) {
            final @Nullable Object dimmerModeConfig = configuration.get(DIMMER_MODE);
```
even though it is checked before it complains about potential NPE. and configuration can be and is "null" for many items. i considered to create an empty configuration for every item but it is imho too much to just get rid of nullable.

personally, i would remove all "suppress" as it could hide real issue introduced later but many false NPE warnings also makes difficult to see real NPEs. so, not sure what is the best approach here. 

suggestions are welcome - should i remove/reduce suppression?